prov/efa/src/efa.h

```diff
@@ -469,4 +469,9 @@ static inline bool efa_ep_is_cuda_mr(struct efa_mr *efa_mr)
 	return efa_mr ? (efa_mr->peer.iface == FI_HMEM_CUDA): false;
 }
 
+static inline bool efa_is_cache_available(struct efa_domain *efa_domain)
+{
+	return efa_domain->cache ? efa_mr_cache_enable : false;
```

Comment:
This check doesn't seem right. As it's written this could return false if `efa_mr_cache_enable` is 0 but `efa_domain->cache` is not NULL. I realize that might not happen but why make it more complicated than it needs to be?

Shouldn't we check `efa_mr_cache_enable` before cache init, and then use `efa_domain->cache` for all other checks in the future? Whether `efa_domain->cache` is set should be the source of truth for whether that domain has a cache, not a global variable.

Reply:
If efa_mr_cache_enable is 0 then we want it to return though, 'cause we specified that we don't want a cache.

Reply:
But we allocated a cache? That makes no sense.

Reply:
I guess it could be:
```
return efa_mr_cache_enable ? efa_domain->cache : false;
```
or
```
return efa_domain->cache ? true : false;
```
It ends up doing the same thing though.

If the cache is not available, return false. If the cache is available return true (efa_mr_cache_enable should be true at this point).

Reply:
I'm failing to see why we need to look at `efa_mr_cache_enable` at all after domain init.

Either `domain->cache` is not NULL and we have a cache, or it's NULL and we don't have a cache.

Reply:
Yeah I changed it to just ```return efa_domain->cache;```

I'm going to leave it as an in-line so it's easier to understand.